File: CoreAngularDemo.API/CoreAngularDemo.DAL/Models/Entities/Supplier.cs
using System;
using System.Collections.Generic;
using CoreAngularDemo.DAL.Models.Entities.Abstractions;

namespace CoreAngularDemo.DAL.Models.Entities
{
    public class Supplier : IAuditableEntity, IBaseEntity
    {
        public Supplier()
        { 
            IssueLog = new HashSet<IssueLog>();
            SupplierParts = new HashSet<SupplierPart>();
        }

        public int Id { get; set; }
        public string Name { get; set; }
        public DateTime? CreatedDate { get; set; }
        public DateTime? UpdatedDate { get; set; }
        public string CreatedById { get; set; }
        public string UpdatedById { get; set; }
        public string FullName { get; set; }
        public int? CountryId { get; set; }
        public int? CurrencyId { get; set; }
        public string Edrpou { get; set; }

        public virtual Country Country { get; set; }
        public virtual Currency Currency { get; set; }
        public virtual User Create { get; set; }
        public virtual User Mod { get; set; }

        public virtual ICollection<IssueLog> IssueLog { get; set; }
        public virtual ICollection<SupplierPart> SupplierParts { get; set; }
    }
}

File: CoreAngularDemo.Client/src/app/modules/shared/models/supplier.ts
import { TEntity } from '../../core/models/entity/entity';
import { Country } from './country';
import { Currency } from './currency';
import { User } from './user';

export class Supplier extends TEntity<Supplier> {
  public name?: string;
  public fullName?: string;
  public edrpou?: string;
  public currency: Currency;
  public currencyFullName?: string;
  public country: Country;
  public countryName?: string;
  public createdDate?: Date;
  public updatedDate?: Date;
  public create?: User;
  public mod?: User;
}
